### PR TITLE
Bebop2 + handshake fix

### PR DESCRIPTION
**Pull request by elpiel**
- Bebop2
- `Drone::connect()` instead of `Drone::new()`
- `Config`
- `send_datetime`
- `pub mod {crate}::prelude`
- `handshake` to own module
- general clean up

### CODE REVIEW

**Comment by o0Ignition0o** [comment]
Looks great already!
I ll try to have a look at it this weekend :)

File: arsdk-rs/src/command.rs [hunk 1]
```diff
@@ -1,11 +1,12 @@
 use crate::common;
 use crate::frame::Data;
 use crate::jumping_sumo;
+use crate::ardrone3::ArDrone3;
 #[derive(Debug, PartialEq, Clone, Copy)]
 
 pub enum Feature {
```
Comment (elpiel) [inline arsdk-rs/src/command.rs:7]: IDK if we should handle the enum like that, but we can definitely figure it out later.
This works for now, although the `ArDrone3` doesn't have all options/class and etc

File: arsdk-rs/src/lib.rs [hunk 2]
```diff
@@ -71,19 +114,21 @@ struct HandshakeResponse {
...
+        // arstream2_client_control_port: Some(44446),
     };
 
     println!(
         "connecting controller {}",
         handshake_message.controller_name,
     );
 
-    let mut handshake_stream = TcpStream::connect(target)?;
+    let mut handshake_stream = retry(10, target).ok_or_else(|| anyhow!("Couldn't connect for handshake {}", target))?;
```
Comment (elpiel) [inline arsdk-rs/src/lib.rs]: We might want to switch to something better (i.e. return a Vec<Errors>)
PS: I am not a big fan of the `anyhow`

File: bebop2/examples/take_off.rs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+use std::error::Error;
+use std::net::IpAddr;
+
+use bebop2::Bebop2;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    // let drone_address = "10.202.0.254".parse::<IpAddr>()?;
+    let drone_address = "10.202.0.1".parse::<IpAddr>()?;
```
Comment (elpiel) [inline bebop2/examples/take_off.rs]: seems to be the correct IP when using sphinx, but the communication is not ok :D

**Comment by o0Ignition0o** [comment]
Ok it looks amazing already, and we're still in the early phase. i m going to merge this, and publish it as is for now !

**Pull request merged**